samples/matter/thermostat/src/temp_sensor_manager.cpp

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "temp_sensor_manager.h"
+#include "app_config.h"
+#include "app_task.h"
+#include "app_event.h"
+
+
+
+
+LOG_MODULE_DECLARE(app, CONFIG_CHIP_APP_LOG_LEVEL);
+
+using namespace chip;
+using namespace chip::DeviceLayer;
+
+
+constexpr EndpointId kThermostatEndpoint = 1;
+
+constexpr uint16_t kSensorTimerPeriodMs  = 30000; // 30s timer period
+constexpr uint16_t kSimulatedReadingFrequency = (60000 / kSensorTimerPeriodMs); // Change Simulated number at each minutes
+
+k_timer sSensorTimer;
+
+#if CONFIG_TEMPERATURE_STEP == 0
+static int16_t mMockTemp[]               = {2000, 
+                                            2200, 
+                                            1600, 
+                                            2200, 
+                                            2125, 
+                                            2999};
+
+#else
+static int16_t mMockedTemperatureStep = CONFIG_TEMPERATURE_STEP;
+#endif 
+
+
+
+
+
+CHIP_ERROR TempSensorManager::Init()
+{
+    // Initialize temp sensor timer
+
+    int16_t InitValue            = 0;
+    
+    PlatformMgr().LockChipStack();
+    app::Clusters::Thermostat::Attributes::LocalTemperature::Set(kThermostatEndpoint, InitValue);
+    PlatformMgr().UnlockChipStack();
+
+    AppEvent event;
+    event.Handler            = TempSensorManager::SensorTimerEventHandler;
+    AppTask::Instance().PostEvent(event);
+
+    k_timer_init(&sSensorTimer, &TempSensorManager::TimerEventHandler, nullptr);
+    k_timer_user_data_set(&sSensorTimer, this);
+    k_timer_start(&sSensorTimer, K_MSEC(kSensorTimerPeriodMs), K_NO_WAIT);
+    return CHIP_NO_ERROR;
+}
+
+
+
+
+
+void TempSensorManager::TimerEventHandler(k_timer * timer)
+{
+    
+    AppEvent event;
+    event.Type               = AppEventType::Timer;
+    event.TimerEvent.Context = k_timer_user_data_get(timer);
+    event.Handler            = TempSensorManager::SensorTimerEventHandler;
+    AppTask::Instance().PostEvent(event);
+  
+}
+
+
+void TempSensorManager::SensorTimerEventHandler(const AppEvent &Event)
+{
+    static int16_t PreviousTemperature = 0;
+
+    #ifdef CONFIG_TEMPERATURE_SENSORS_MOCKED
+
+
+    static uint8_t MockTempIndex = 0;
+    static uint8_t CycleCounter = 0;
+    
+    #if CONFIG_TEMPERATURE_STEP == 0
+    
+    int16_t Temperature            = 0;
+
+
+    Temperature = mMockTemp[MockTempIndex];
+
+    CycleCounter++;
+    if (CycleCounter >= kSimulatedReadingFrequency)
+    {
+        if(MockTempIndex >= ArraySize(mMockTemp) - 1){
```

Comment:
Is this condition correct? 
```suggestion
        if(MockTempIndex > ArraySize(mMockTemp) - 1){
```